packages/apps/src/app.plugins.ts

```diff
@@ -29,17 +30,26 @@ export function plugin(this: App, plugin: IPlugin) {
 /**
  * get a plugin
  */
-export function getPlugin(this: App, name: PluginName): IPlugin | undefined {
+export function getPlugin<TPlugin extends IPlugin>(
+  this: App<TPlugin>,
+  name: PluginName
+): IPlugin | undefined {
   return this.plugins.find((plugin) => {
     const metadata = getMetadata(plugin);
     return metadata.name === name;
   });
 }
 
+type TupleUnion<U extends string, R extends any[] = []> = {
```

Comment:
we need a type-helper file

Reply:
if you could move all these "utility types" to the `./types` directory like you see with this one https://github.com/microsoft/teams.ts/blob/main/packages/apps/src/types/prefixed.ts that would be great and keep the repo tidy.